server/server.go

```diff
@@ -321,6 +332,12 @@ func (a *ArgoCDServer) Run(ctx context.Context, port int, metricsPort int) {
 
 	go a.projInformer.Run(ctx.Done())
 	go a.appInformer.Run(ctx.Done())
+	go a.nodeInformer.Run(ctx.Done())
+
+	if !cache.WaitForCacheSync(ctx.Done(), a.nodeInformer.HasSynced) {
```

Comment:
There is `cache.WaitForCacheSync` for other informers at line 351. Please add a new informer there.